main.tf

```diff
@@ -209,6 +210,24 @@ data "aws_iam_policy_document" "resource" {
 
 resource "aws_ecr_repository_policy" "name" {
   for_each   = toset(local.ecr_need_policy && module.this.enabled ? local.image_names : [])
-  repository = aws_ecr_repository.name[each.value].name
+  repository = each.value
```

Comment:
Unsure about this change, but easiest way to find out is to test via the component plan.